### PR TITLE
Changed order of handle-route (update before create)

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

The order of update and create route is important if you want to create an article with an existing route.

Additional changes:

* added `route_path` to create
* fixed delete selected in list

### CODE REVIEW

File: Document/Index/ArticleIndexer.php [hunk 1]
```diff
@@ -200,29 +198,14 @@ protected function createOrUpdateArticle(
...
         $article->setAuthored($document->getAuthored());
-        if ($document->getAuthor()) {
-            /** @var Contact $author */
-            $author = $this->contactRepository->findById($document->getAuthor());
-            if ($author) {
-                $article->setAuthorFullName($author->getFullName());
-                $article->setAuthorId($author->getId());
-            }
+        if ($document->getAuthor() && $author = $this->contactRepository->findById($document->getAuthor())) {
+            $article->setAuthorFullName($author->getFullName());
```
Comment (trickreich) [inline Document/Index/ArticleIndexer.php]: @wachterjohannes Why you are not setting the author id anymore?

Comment (wachterjohannes) [inline Document/Index/ArticleIndexer.php]: i have only written it in another format - to make it more readable.

Comment (trickreich) [inline Document/Index/ArticleIndexer.php]: You have changed the function...

Comment (trickreich) [inline Document/Index/ArticleIndexer.php]: Following line is missing:
```
$article->setAuthorId($author->getId());
```

File: Document/Index/ArticleIndexer.php [hunk 2]
```diff
@@ -200,29 +198,14 @@ protected function createOrUpdateArticle(
...
         }
-        if ($document->getChanger()) {
-            /** @var User $changer */
-            $changer = $this->userManager->getUserById($document->getChanger());
-            if ($changer) {
-                $article->setChangerFullName($changer->getFullName());
-                $article->setChangerContactId($changer->getContact()->getId());
-            }
+        if ($document->getChanger() && $changer = $this->userManager->getUserById($document->getChanger())) {
+            $article->setChangerFullName($changer->getFullName());
```
Comment (trickreich) [inline Document/Index/ArticleIndexer.php]: same here

File: Document/Index/ArticleIndexer.php [hunk 3]
```diff
@@ -200,29 +198,14 @@ protected function createOrUpdateArticle(
...
         }
-        if ($document->getCreator()) {
-            /** @var User $creator */
-            $creator = $this->userManager->getUserById($document->getCreator());
-            if ($creator) {
-                $article->setCreatorFullName($creator->getFullName());
-                $article->setCreatorContactId($creator->getContact()->getId());
-            }
+        if ($document->getCreator() && $creator = $this->userManager->getUserById($document->getCreator())) {
+            $article->setCreatorFullName($creator->getFullName());
```
Comment (trickreich) [inline Document/Index/ArticleIndexer.php]: same here

File: Document/Index/ArticleIndexer.php (hunk 2 above)
Comment (wachterjohannes) [inline Document/Index/ArticleIndexer.php]: same here (:

File: Document/Index/ArticleIndexer.php (hunk 3 above)
Comment (wachterjohannes) [inline Document/Index/ArticleIndexer.php]: same here (:

**Comment by wachterjohannes** [comment]
@trickreich fixed.

File: Document/Subscriber/ArticleSubscriber.php [hunk 4]
```diff
@@ -102,7 +102,7 @@ public static function getSubscribedEvents()
     {
         return [
             Events::HYDRATE => [['handleHydrate', -500]],
-            Events::PERSIST => [['handleRoute', 0], ['handleRouteUpdate', 0], ['handleScheduleIndex', -500]],
+            Events::PERSIST => [['handleRouteUpdate', 1], ['handleRoute', 0], ['handleScheduleIndex', -500]],
```
Comment (trickreich) [inline Document/Subscriber/ArticleSubscriber.php]: I thought we make here bigger steps..

File: Resources/public/js/components/articles/list/main.js [hunk 5]
```diff
@@ -277,7 +277,10 @@ define([
         },
 
         deleteItems: function(ids) {
-            this.sandbox.util.save('/admin/api/articles?ids=' + ids.join(','), 'DELETE').then(function() {
+            this.sandbox.util.save(
```
Comment (trickreich) [inline Resources/public/js/components/articles/list/main.js]: We should use here the ArticleManager

File: Document/Subscriber/ArticleSubscriber.php (hunk 4 above)
Comment (wachterjohannes) [inline Document/Subscriber/ArticleSubscriber.php]: it is here important that this is near to each other and in the right order. if you use a bigger priority you have no route - if you use a smaller you have one.

**Comment by wachterjohannes** [comment]
@trickreich fixed

**Comment by trickreich** [comment]
ack

**Pull request merged**